x-pack/plugin/identity-provider/src/main/java/org/elasticsearch/xpack/idp/saml/idp/CloudIdp.java

```diff
@@ -0,0 +1,121 @@
+        this.sloEndpoints.computeIfAbsent("redirect", v -> settings.get(IDP_SLO_REDIRECT_ENDPOINT.getKey()));
+        this.signingCredential = buildSigningCredential(env, settings);
+    }
+
+    @Override
+    public String getEntityId() {
+        return entityId;
+    }
+
+    @Override
+    public String getSingleSignOnEndpoint(String binding) {
+        return ssoEndpoints.get(binding);
+    }
+
+    @Override
+    public String getSingleLogoutEndpoint(String binding) {
+        return sloEndpoints.get(binding);
+    }
+
+    @Override
+    public X509Credential getSigningCredential() {
+        return signingCredential;
+    }
+
+    private static String require(Settings settings, Setting<String> setting) {
+        if (settings.hasValue(setting.getKey())) {
+            return setting.get(settings);
+        } else {
+            throw new SettingsException("The configuration setting [" + setting.getKey() + "] is required");
+        }
+    }
+
+    static X509Credential buildSigningCredential(Environment env, Settings settings) {
+        final X509KeyPairSettings keyPairSettings = X509KeyPairSettings.withPrefix("xpack.idp.signing.", false);
+        final X509KeyManager keyManager = CertParsingUtils.getKeyManager(keyPairSettings, settings, null, env);
+        if (keyManager == null) {
+            return null;
+        }
+        final Set<String> aliases = new HashSet<>();
+        final String configAlias = IDP_SIGNING_KEY_ALIAS.get(settings);
+        if (Strings.isNullOrEmpty(configAlias)) {
+            final String[] rsaAliases = keyManager.getServerAliases("RSA", null);
+            if (null != rsaAliases) {
+                aliases.addAll(Arrays.asList(rsaAliases));
+            }
+            final String[] ecAliases = keyManager.getServerAliases("EC", null);
+            if (null != ecAliases) {
+                aliases.addAll(Arrays.asList(ecAliases));
+            }
+            if (aliases.isEmpty()) {
+                throw new IllegalArgumentException("The configured keystore for xpack.idp.signing.keystore does not contain any RSA or EC" +
+                    " key pairs");
+            }
+            if (aliases.size() > 1) {
+                throw new IllegalArgumentException("The configured keystore for xpack.idp.signing.keystore contains multiple key pairs" +
+                    " but no alias has been configured with [" + IDP_SIGNING_KEY_ALIAS.getKey() + "]");
+            }
+        } else {
+            aliases.add(configAlias);
+        }
+        String alias = new ArrayList<>(aliases).get(0);
+        if (keyManager.getPrivateKey(alias) == null) {
+            throw new IllegalArgumentException("The configured keystore for xpack.idp.signing.keystore does not have a private key" +
+                " associated with alias [" + alias + "]");
+        }
+
+        final String keyType = keyManager.getPrivateKey(alias).getAlgorithm();
```

Comment:
Minor: Can we extract `keyManager.getPrivateKey(alias)` into a local since we call it twice?